### PR TITLE
Generate comments

### PR DESCRIPTION
**Pull request by kvigen**
This change fixes some of the golinting issues our generated code. For the most part it does this by adding comments. It also makes a few functions private, though when it was easy I added comments for those too.

### CODE REVIEW

File: models/genmodels.go [hunk 1]
```diff
@@ -222,8 +230,13 @@ func generateSuccessTypes(capOpID string, responses map[int]spec.Response) (stri
 		if err != nil {
 			return "", err
 		}
+		buf.WriteString(fmt.Sprintf("// %s defines the %d status code response for %s\n",
```
Comment (kvigen) [inline models/genmodels.go]: i may try to move this to use templates

File: generated/client/client.go [hunk 2]
```diff
@@ -16,8 +16,9 @@ var _ = strings.Replace
 var _ = strconv.FormatInt
 var _ = bytes.Compare
 
+// Client makes requests to Swagger Test
```
Comment (rgarcia) [inline generated/client/client.go]: nit: "to the Swagger Test service."

File: generated/client/client.go [hunk 3]
```diff
@@ -60,8 +63,10 @@ func JoinByFormat(data []string, format string) string {
 	}
 	return strings.Join(data, sep)
 }
+
+// GetBooks returns...
```
Comment (rgarcia) [inline generated/client/client.go]: Maybe `GetBooks makes a <GET/POST/...> request to <URL pattern from swagger spec>. <Description field from swagger spec>`.

File: generated/models/inputs.go [hunk 4]
```diff
@@ -13,6 +13,7 @@ var _ = strconv.FormatInt
 var _ = validate.Maximum
 var _ = strfmt.NewFormats
 
+// GetBooksInput holds the input parameters for a getBooks operation
```
Comment (rgarcia) [inline generated/models/inputs.go]: nit: add period

File: generated/models/inputs.go [hunk 5]
```diff
@@ -24,6 +25,8 @@ type GetBooksInput struct {
 	PagesToTime *float32
 }
 
+// Validate returns an error if any of the GetBooksInput parameters don't satisfy the
+// requirements from the swagger yml file
```
Comment (rgarcia) [inline generated/models/inputs.go]: nit: period

Comment (rgarcia) [inline generated/models/inputs.go]: same for all other comments

File: generated/models/outputs.go [hunk 6]
```diff
@@ -5,81 +5,91 @@ type DefaultInternalError struct {
 	Msg string `json:"msg"`
 }
 
+// Error returns the internal error that caused the 500
 func (d DefaultInternalError) Error() string {
 	return d.Msg
 }
 
+// DefaultBadRequest represents a generic 400 response
```
Comment (rgarcia) [inline generated/models/outputs.go]: Could expand on this to explain how it's used by wag internally.

File: generated/server/interface.go [hunk 7]
```diff
@@ -7,9 +7,14 @@ import (
 
 //go:generate $GOPATH/bin/mockgen -source=$GOFILE -destination=mock_controller.go -package=server
 
+// Controller defines the interface for Swagger Test
```
Comment (rgarcia) [inline generated/server/interface.go]: "for the Swagger Test service."

File: generated/server/router.go [hunk 8]
```diff
@@ -27,6 +30,7 @@ type handler struct {
 	Controller
 }
 
+// New returns a Server that implements the Controller interface. It will start "Serve" is called.
```
Comment (rgarcia) [inline generated/server/router.go]: start *when* Server is called

**Comment by rgarcia** [comment]
lgtm besides some nits

File: generated/models/inputs.go (hunk 4 above)
Comment (kvigen) [inline generated/models/inputs.go]: 👍

File: generated/client/client.go (hunk 3 above)
Comment (kvigen) [inline generated/client/client.go]: i like that

**Pull request merged**